features/step_definitions/step_eight_steps.rb

```diff
@@ -3,45 +3,36 @@ def group_eight(index)
 end
 
 Given(/^I am a single person on step eight$/) do
-  step 'I visit the page for step one'
-  step 'I enter a valid form number'
+  step 'I am on the page for step one'
+  step 'I submit the form with a valid form number'
   step 'I select no to have you already paid the fee?'
-  step 'I select single'
-  step 'I select £0 to £2,999'
-  step 'I select no to do you receive any of the following benefits?'
-  step 'I select no to do you have any children'
+  step 'I click continue'
+  step 'I submit the form as single'
+  step 'I submit the form with £0 to £2,999 checked'
+  step 'I submit the form with no I do not receive one of the benefits listed'
+  step 'I submit the form with no I do not have any children'
+  expect(step_eight_page.content).to have_choose_income_single
 end
 
 Given(/^I am a married person on step eight$/) do
-  step 'I visit the page for step one'
-  step 'I enter a valid form number'
+  step 'I am on the page for step one'
+  step 'I submit the form with a valid form number'
   step 'I select no to have you already paid the fee?'
-  step 'I select married'
-  step 'I select £0 to £2,999'
-  step 'I select no to do you receive any of the following benefits?'
-  step 'I select no to do you have any children'
+  step 'I click continue'
+  step 'I submit the form as married'
+  step 'I submit the form with £0 to £2,999 checked'
+  step 'I submit the form with no I do not receive one of the benefits listed'
+  step 'I submit the form with no I do not have any children'
+  expect(step_eight_page.content).to have_choose_income_married
 end
 
-When(/^I select wages from income list$/) do
-  expect(group_eight(0).block_label[0].text).to eq 'Wages'
-  group_eight(0).block_label[0].click
-end
-
-And(/^I select working tax credit from income list$/) do
-  expect(group_eight(0).block_label[2].text).to eq 'Working Tax Credit'
-  group_eight(0).block_label[2].click
-  common_page.continue_button.click
-end
-
-When(/^I select no income$/) do
-  group_eight(0).input[12].click
+When(/^I submit the form with no income checked$/) do
+  group_eight(0).no_income.click
```

Comment:
I wondered before what this group_{number}({another_number}) means ? it does not read well at all imo.  Can this be defined in a different way to make it more readable ?

Reply:
I'm sure there is, would appreciate your help on this but will continue to refactor on a new branch